share/completions/git.fish

```diff
@@ -1956,10 +1956,10 @@ complete -F -c git -n '__fish_git_using_command restore' -n '__fish_git_contains
 # switch options
 complete -f -c git -n __fish_git_needs_command -a switch -d 'Switch to a branch'
 complete -f -c git -n '__fish_git_using_command switch' -ka '(__fish_git_unique_remote_branches)' -d 'Unique Remote Branch'
-complete -f -c git -n '__fish_git_using_command switch' -ka '(__fish_git_local_branches)'
+complete -f -c git -n '__fish_git_using_command switch' -ka '(__fish_git_branches)'
 complete -f -c git -n '__fish_git_using_command switch' -s c -l create -d 'Create a new branch'
 complete -f -c git -n '__fish_git_using_command switch' -s C -l force-create -d 'Force create a new branch'
-complete -f -c git -n '__fish_git_using_command switch' -s d -l detach -d 'Switch to a commit for inspection and discardable experiment' -rka '(__fish_git_refs)'
+complete -f -c git -n '__fish_git_using_command switch' -s d -l detach -d 'Switch to a commit for inspection and discardable experiment'
```

Comment:
This should be kept. Giving commits to `switch --detach` is okay.

Reply:
Since the `--detach` can come after the commit (e.g., `switch <start-point> --detach`), I'll modify argument completion to use `__fish_git_refs` instead of `__fish_git_branches` because `_fish_git_refs` includes `__fish_git_branches`.

Reply:
No. We complete what we know, so we complete refs if we got `--detach`.

Sometimes offering less is more. Keeping branches otherwise is fine!

Reply:
The issue is completing random commits isn't very useful here, and if you do `git switch 374637deadbeef` you'll then *have to* add `--detach`.

It's *fine* if that's not modelled 100% - just keep it simple and offer the much more useful branches unless `--detach` is already given, then switch to offering all "refs".

Reply:
Ahh — so for the form `switch --detach <starting-point>` to use `__fish_git_refs` will result in that form to get all references.

I'll make that adjustment now.

Reply:
I've updated the PR to bring back `__fish_git_refs` for `--detach` and updated the commit message appropriately.